### PR TITLE
Add a CODEOWNERS file

### PR DESCRIPTION
**Pull request by iHiD**
Add a CODEOWNERS file to protect merges to the Dockerfile from anyone who is not part of the @exercism/ops.

### CODE REVIEW

**Comment by iHiD** [comment]
@kytrinyx So the main concern I have with this is that @ErikSchierboom won't be able to merge his own PRs now, as they require a review. Thoughts?

**Comment by kytrinyx** [comment]
> So the main concern I have with this is that @ErikSchierboom won't be able to merge his own PRs now, as they require a review.

Eeew, yeah that's no good. In that case, let's just assume Erik won't change the CODEOWNERS and Dockerfile files and push directly to master... :-) (I mean, I actually do trust that this is the case!)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+/Dockerfile @exercism/ops
```
Comment (ErikSchierboom) [inline .github/CODEOWNERS]: I think the `.dockerignore` file should also be added. And what about the `CODEOWNERS` file itself?

**Comment by iHiD** [comment]
So we have a CODEOWNERS file for pinging us, but not for protection?

So this is going to be an issue for any place where there is only one maintainer though (which is currently all the SA places). We need to decide (internally) if we're happy with the risk-profile of this or if we need to take an alternative route.

I'll do that for this repo for now though :)

**Comment by ErikSchierboom** [comment]
> So we have a CODEOWNERS file for pinging us, but not for protection?

Hmmm, that's too bad :(

**Comment by kytrinyx** [comment]
> So we have a CODEOWNERS file for pinging us, but not for protection?

Yepp.

**Comment by iHiD** [comment]
So, one solution would be:
- Leave off "Requires review"
- Turn on "Requires status checks"
- Add a status checking bot thingymagig, which fails if Dockerfile has changed, unless there has been a review from a member of `@exercism/ops`.

That's going to involve more coding but I think it **would** get us around the problem. Thoughts?

File: .github/CODEOWNERS (hunk 1 above)
Comment (iHiD) [reply .github/CODEOWNERS]: Why should .dockerignore be added?

Comment (iHiD) [reply .github/CODEOWNERS]: I now know this from your PR to the Ruby one. Thanks :)

Comment (ErikSchierboom) [reply .github/CODEOWNERS]: :)

Comment (ErikSchierboom) [reply .github/CODEOWNERS]: Should I merge or hold out on this?

**Comment by ErikSchierboom** [comment]
So shall I merge?

**Comment by iHiD** [comment]
Yes please. I'm comfortable too but very much with this being an exception, not a rule.

It's worth being clear that although the Docker containers that do the analysis are in a contained, safe environment, this Dockerfile is run in an effectively uncontrolled environment with full access to network etc, so a malicious addition to this file could be very bad news (it could run up very large bills for us very quickly, or spam people, for example). We therefore need to have a very high level of trust in an individual to allow them to have control over this file.

**Comment by ErikSchierboom** [comment]
Merged!

**Pull request merged**